src/source/Ice/IceAgent.c

```diff
@@ -1774,6 +1774,59 @@ STATUS iceAgentCheckConnectionStateSetup(PIceAgent pIceAgent)
     return retStatus;
 }
 
+STATUS updateCandidateStats(PIceAgent pIceAgent, BOOL isRemote)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    CHK(pIceAgent != NULL && pIceAgent->pDataSendingIceCandidatePair != NULL, STATUS_NULL_ARG);
+    PIceCandidate pIceCandidate = pIceAgent->pDataSendingIceCandidatePair->remote;
+    PRtcIceCandidateDiagnostics pRtcIceCandidateDiagnostics = &pIceAgent->rtcSelectedRemoteIceCandidateDiagnostics;
+    if (!isRemote) {
+        pIceCandidate = pIceAgent->pDataSendingIceCandidatePair->local;
+        pRtcIceCandidateDiagnostics = &pIceAgent->rtcSelectedLocalIceCandidateDiagnostics;
+        // URL and relay protocol are populated only for local candidate by spec
+        STRNCPY(pRtcIceCandidateDiagnostics->url, pIceAgent->iceServers[pIceCandidate->iceServerIndex].url,
+                ARRAY_SIZE(pRtcIceCandidateDiagnostics->url));
+
+        // Only if candidate is obtained from TURN server will relay protocol be populated. Else, relay protocol is
+        // not applicable.
+        STRNCPY(pRtcIceCandidateDiagnostics->relayProtocol, STATS_NOT_APPLICABLE_STR, ARRAY_SIZE(pRtcIceCandidateDiagnostics->relayProtocol));
+        if (pIceCandidate->iceCandidateType == ICE_CANDIDATE_TYPE_RELAYED && pIceCandidate->pTurnConnection != NULL) {
+            switch (pIceCandidate->pTurnConnection->protocol) {
+                case KVS_SOCKET_PROTOCOL_UDP:
+                    STRNCPY(pRtcIceCandidateDiagnostics->relayProtocol, ICE_URL_TRANSPORT_UDP,
+                            ARRAY_SIZE(pRtcIceCandidateDiagnostics->relayProtocol));
+                    break;
+                case KVS_SOCKET_PROTOCOL_TCP:
+                    STRNCPY(pRtcIceCandidateDiagnostics->relayProtocol, ICE_URL_TRANSPORT_TCP,
+                            ARRAY_SIZE(pIceAgent->rtcSelectedLocalIceCandidateDiagnostics.relayProtocol));
+                    break;
+                default:
+                    MEMSET(pRtcIceCandidateDiagnostics->relayProtocol, 0, SIZEOF(pRtcIceCandidateDiagnostics->relayProtocol));
+            }
+        }
+    }
+    getIpAddrStr(&pIceCandidate->ipAddress, pRtcIceCandidateDiagnostics->address, ARRAY_SIZE(pRtcIceCandidateDiagnostics->address));
+    pRtcIceCandidateDiagnostics->port = pIceCandidate->ipAddress.port;
```

Comment:
pIceCandidate->ipAddress.port is in network byte order though. need to convert?

Reply:
Done